### PR TITLE
Talk page voiceover improvements

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** https://phabricator.wikimedia.org/T318611

### Notes
* This PR adds voiceover support to Talk Pages. The goal is to provide better context to some elements on the screen and adjusts the focus when necessary.
* Feel free to suggest text changes that would provide better context to users

### Test Steps
1. With voiceover on, navigate the talk pages, try to post and reply to topics, and simulate errors to check the error feedback.

### CODE REVIEW

File: Wikipedia/Code/TalkPageTopicReplyOnboardingView.swift [hunk 1]
```diff
@@ -55,6 +58,7 @@ struct TalkPageTopicReplyOnboardingView: View {
             }
             ZStack(alignment: .bottom, content: {
                 VStack {
+                    if #available(iOS 14.0, *) { // TODO: remove version check
```
Comment (mazevedofs) [inline Wikipedia/Code/TalkPageTopicReplyOnboardingView.swift:61]: I didn't add fallback as we're dropping iOS 13 in this version

**Review by tonisevener — CHANGES_REQUESTED** [review]
Sounding great! I have some thoughts on more items that could be accessible, but I think they make more sense to add them as a follow-on PR. So this is feedback just for the items tackled in this PR.

File: WMF Framework/CommonStrings.swift [hunk 2]
```diff
@@ -262,7 +262,12 @@ public class CommonStrings: NSObject {
     public static func talkPageReply(languageCode: String?) -> String {
         WMFLocalizedString("talk-page-reply-button", languageCode: languageCode, value: "Reply", comment: "Text used on button to reply to talk page messages.  Please prioritize for de, ar and zh wikis.")
     }
-    
+    @objc public static let talkPageReplyAccessibilityText = WMFLocalizedString("talk-page-reply-button-accessibility-label", value: "Reply to %@", comment: "Accessibility text for repy button. The %@ will be replaced with the name of the user whose comment is being responded")
+
+    public static let enabledAccessibilityLabel = WMFLocalizedString("talk-page-reply-button-enabled", value: "Enabled", comment: "Accessibility text indicating that the button is enabled")
```
[the remark this answers is not in the document]
Comment (mazevedofs) [reply WMF Framework/CommonStrings.swift:267]: Good catch, I hadn't noticed this before! Maybe I was moving on between elements too fast. I'll remove those

File: Wikipedia/Code/TalkPageCell.swift [hunk 3]
```diff
@@ -163,6 +159,11 @@ final class TalkPageCell: UICollectionViewCell {
                 
                 stackView.addArrangedSubview(separator)
                 stackView.addArrangedSubview(commentView)
+
+                let replyButtonAccessibilityLabel = CommonStrings.talkPageReplyAccessibilityText
```
[the remark this answers is not in the document]
Comment (mazevedofs) [reply Wikipedia/Code/TalkPageCell.swift:163]: No reason 😂 , I'll move it out of the loop

File: Wikipedia/Code/TalkPageViewController.swift [hunk 4]
```diff
@@ -920,8 +928,10 @@ extension TalkPageViewController: UIAdaptivePresentationControllerDelegate {
         guard topicComposeVC.shouldBlockDismissal else {
             return true
         }
-        
-        topicComposeVC.presentDismissConfirmationActionSheet()
+
+        if !UIAccessibility.isVoiceOverRunning {
```
[the remark this answers is not in the document]
Comment (mazevedofs) [reply Wikipedia/Code/TalkPageViewController.swift:932]: It wasn't reading it out loud, but it worked with breakpoints. That's why I added a way to "delay" it

**Comment by mazevedofs** [comment]
@tonisevener I guess it's ready for another review!

**Review by tonisevener — APPROVED** [review]
Working great!

**Pull request merged**